File: solutions/cpp/pangram/5/pangram.cpp
#include "pangram.h"

namespace pangram {

bool is_pangram(const std::string& input)
{
    unsigned int pangram_map {};
    for (auto ch: input)
    {
        if ('a' <= ch && ch <= 'z')
        {
            pangram_map |= (1u << (ch - 'a')); 
        }
        if ('A' <= ch && ch <= 'Z')
        {
            pangram_map |= (1u << (ch - 'A'));
        }
    }
    return __builtin_popcount(pangram_map) == 26;
}

}  // namespace pangram

File: solutions/javascript/pangram/3/pangram.js
const PANGRAM = 0x3FFFFFF
const charIsLower = ch => ch >= 'a' && ch <= 'z'
const charIdx = ch => ch.charCodeAt(0) - 'a'.charCodeAt(0)
export const isPangram = input => PANGRAM === Array.from(input.toLowerCase())
  .reduce((bits, ch) => !charIsLower(ch) ? bits : bits | 1 << charIdx(ch), 0)